CHANGELOG.md

```diff
@@ -38,6 +38,10 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
     Inner items of inline tables are no longer merged across configuration
     files.
 
+* `jj abandon` now deletes bookmarks pointing to the revisions to be abandoned.
+  Use `--retain-bookmarks` to move bookmarks backwards.
+  [https://github.com/jj-vcs/jj/issues/3505](#3505)
```

Comment:
This might be the most important breaking change, so put it first? (We sometime reorder items when we do the release, but sometimes we forget.)

Reply:
I wonder if we should mention somewhere that a subsequent `jj git push --all` will delete the branch from the remote. Do you think a note here is good? And/or in `jj abandon --help`?

Reply:
Updated the changelog entry. I don't think it's surprising that deleted bookmarks are propagated on push, so I think changelog entry is good.